app/src/main/java/com/asma/tasky/feature_authentication/presentation/register/RegisterViewModel.kt

```diff
@@ -0,0 +1,146 @@
+package com.asma.tasky.feature_authentication.presentation.register
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.asma.tasky.R
+import com.asma.tasky.feature_authentication.domain.util.AuthError
+import com.asma.tasky.feature_authentication.domain.util.Validation
+import com.asma.tasky.core.util.Resource
+import com.asma.tasky.core.domain.states.TextFieldState
+import com.asma.tasky.core.presentation.util.UiEvent
+import com.asma.tasky.core.util.UiText
+import com.asma.tasky.feature_authentication.domain.repository.AuthenticationRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class RegisterViewModel @Inject constructor(
+    private val repository: AuthenticationRepository
+) : ViewModel() {
+
+    private val _nameState = mutableStateOf(TextFieldState())
+    val nameState: State<TextFieldState> = _nameState
```

Comment:
Any reason you use compose states in this ViewModel and StateFlows in the other one?
I'd personally use StateFlows though, since they allow you to better transform certain states due to all the cool flow operators. Also it's a good idea to keep ViewModels Compose-free to allow them to be locally unit testable and reusable.

And if you still decide to use Compose states in your VM, you can get rid of using 2 states for each. Instead you can do this:
```
var nameState by mutableStateOf(TextFieldState())
    private set
```